packages/components/src/MultiSelect/subcomponents/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useEffect, useRef, InputHTMLAttributes } from "react"
+import classnames from "classnames"
+import { CheckIcon } from "~icons/CheckIcon"
+import { MinusIcon } from "~icons/MinusIcon"
+import { OverrideClassName } from "~types/OverrideClassName"
+import styles from "./Checkbox.module.scss"
+
+export type CheckedStatus = "checked" | "unchecked" | "indeterminate"
+
+type ReadOnly = {
+  readOnly: true
+  onChange?: never
+}
+
+type Controlled = {
+  readOnly?: false
+  onChange: Required<InputHTMLAttributes<HTMLInputElement>["onChange"]>
+}
+
+export type CheckboxProps = {
+  checkedStatus: CheckedStatus
```

Comment:
Do we want this to match the native attribute? As spoken about recently with icons.

If we're supporting mixed I guess it would be `aria-checked`.

I question whether we should be supporting indeterminate states though.

Reply:
We wanted to avoid `aria-checked` since we're leveraging as much of the native `input` as possible, so felt a little like a hat on a hat. We did validate that setting indeterminate in ln55 does get read to the screen reader as expected if that helps for context
![Screenshot 2023-09-04 at 3 05 55 pm](https://github.com/cultureamp/kaizen-design-system/assets/36558508/74d6a511-9cf3-428e-abe4-f6cfa54ddc8b)

Reply:
In terms of feature parity for the current checkbox and how its used in the platform, I think we do have indeterminate use cases (ie: the goals UI example recently shared). But worth asking our design fellows whether this is something we want to cater for

Reply:
Yeah I kinda just wonder if it's one of those things like `<select multiple>` where yes it's (kinda) supported by HTML but how many people actually know what it means when there's a minus in a checkbox?

Plus slight concern for support with different screen reader + browser combos. Bit out of date and doesn't cover JAWS but https://github.com/mui/material-ui/issues/20476#issuecomment-611468844

Reply:
`indeterminate` has been around for a long time, and I think it might have more visual association than you'd guess.

Nonetheless, you're probably right about screen reader + browser combos being a concern, but perhaps that's something that the product teams can alleviate with more aria attributes when they're actually building something? 🤔 

It's not something we need immediately, but I think it's commoncase enough that we can't just completely ignore it unless there is a recommended better solution for it.

Reply:
There might be use cases in auth as well. Regardless, this component will go through a bit of a review before being released as a `future` component to replace existing `Checkbox` so I think it's okay for this API to be somewhat beta/focussed on solving `MultiSelect` specifically.